braindecode/models/util.py

```diff
@@ -1,13 +1,55 @@
 # Authors: Robin Schirrmeister <robintibor@gmail.com>
 #          Hubert Banville <hubert.jbanville@gmail.com>
+#          Bruno Aristimunha <b.aristimunha@gmail.com>
 #
 # License: BSD (3-clause)
+from warnings import warn
 
 import torch
 import numpy as np
 from scipy.special import log_softmax
 
 
+def check_deprecation_warning(self: object, old_new_name_pairs: dict, **kwargs: dict) -> dict:
+    """
+    Check if the user is using a deprecated parameter name and warn him/her
+    about it.
+
+    Parameters
+    ----------
+    self: object
+        The object that contains the deprecated parameters.
+    kwargs: dict
+        A dictionary containing the parameters of the object.
+    old_new_name_pairs: dict
+        A dictionary containing the old parameter name as key and the new
+        parameter name as value.
+
+    Notes
+    -----
+    This function should be called in the __init__ method of the class that
+    contains the deprecated parameters.
+
+    """
+    if not all([k in old_new_name_pairs for k in kwargs]) and \
```

Comment:
I feel better make this in a loop and then give a specific error, like which one is not valid... or you could also use set intersection etc instead of loop, but definitely feel better if error warning says which one is not correct